app/src/main/java/org/fossasia/susi/ai/adapters/viewholders/MapViewHolder.java

```diff
@@ -20,18 +31,76 @@
     @BindView(R.id.map_image)
     public ImageView mapImage;
     @BindView(R.id.location_pointer)
-    public  ImageView pointer;
+    public ImageView pointer;
     @BindView(R.id.background_layout)
     public LinearLayout backgroundLayout;
 
+    private String TAG = ChatFeedRecyclerAdapter.class.getSimpleName();
+
     /**
      * Instantiates a new Map view holder.
      *
      * @param itemView the item view
      * @param listener the listener
      */
-    public MapViewHolder(View itemView , ClickListener listener) {
-        super(itemView , listener);
+    public MapViewHolder(View itemView, ClickListener listener) {
+        super(itemView, listener);
         ButterKnife.bind(this, itemView);
     }
+
+    /**
+     * Inflate MapView
+     *
+     * @param model the ChatMessage object
+     * @param currContext the Context
+     */
+    public void setView(final ChatMessage model, final Context currContext) {
+        if (model != null) {
+            try {
+                final MapHelper mapHelper = new MapHelper(new MapData(model.getLatitude(), model.getLongitude(), model.getZoom()));
+                pointer.setVisibility(View.GONE);
+                Log.v(TAG, mapHelper.getMapURL());
+
+                Picasso.with(currContext).load(mapHelper.getMapURL())
+                        .into(mapImage, new com.squareup.picasso.Callback() {
+                            @Override
+                            public void onSuccess() {
+                                pointer.setVisibility(View.VISIBLE);
+                            }
+
+                            @Override
+                            public void onError() {
+                                Log.d("Error", "map image can't loaded");
+                            }
+                        });
+
+                mapImage.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        /*
+                          Open in Google Maps if installed, otherwise open browser.
+                        */
+                        if (((MainActivity)currContext).recyclerAdapter.selectedItems.size() != 0) {
```

Comment:
pass adapter as param and use adapter.selectedItems() instead of `((MainActivity)currContext).recyclerAdapter.selectedItems.size()`. Do this everywhere.